src/smtencoding/FStar.SMTEncoding.Term.fs

```diff
@@ -717,21 +717,21 @@ let termToSmt
       in
       aux 0 0 [] t
 
-let caption_to_string = function
+let caption_to_string print_captions =
+    function
     | Some c
-       when Options.log_queries() &&
-            Options.keep_query_captions() ->
+       when print_captions ->
         let c = String.split ['\n'] c |> List.map BU.trim_string |> String.concat " " in
         ";;;;;;;;;;;;;;;;" ^ c ^ "\n"
     | _ -> ""
 
 
-let rec declToSmt' print_ranges z3options decl =
+let rec declToSmt' print_captions z3options decl =
   match decl with
   | DefPrelude ->
     mkPrelude z3options
   | Module (s, decls) ->
-    let res = List.map (declToSmt' print_ranges z3options) decls |> String.concat "\n" in
+    let res = List.map (declToSmt' print_captions z3options) decls |> String.concat "\n" in
     if Options.log_queries ()
```

Comment:
Should this be Options.log_queries () && Options.keep_query_captions ()? Else the query hash will depend on the string below (if log_queries is on and keep_query_options is off).